tests/ashley/test_templatetags_custom_tags.py

```diff
@@ -202,26 +220,46 @@ def get_html_post_by(user):
         response = self.client.get(
             reverse("forum:forum", kwargs={"slug": forum.slug, "pk": forum.pk})
         )
-        # check that the topic creation writer is still from our instructor and has the icon
-        self.assertIn(
-            get_html_topic_by(user1),
-            self._trim(response.content.decode("utf-8")),
-        )
+
+        with self.assertHTML(response, ".topic-created") as (html,):
+            extract = parse_html(str(etree.tostring(html)))
+            # control that the instructor's icon is present
+            self.assertTrue(
+                parse_html(
+                    '<i class="icon_writer fas fa-award" aria-hidden="true" title="Instructor">'
+                    '<span class="sr-only">Instructor</span>'
+                    "</i>"
+                )
+                in extract
+            )
+            # control that it's the right profil link
+            self.assertTrue(
+                parse_html(f'<a href="/forum/member/profile/{user1.id}/">Valéry</a>')
+                in extract
+            )
+
         # check that the last post creation writer is not the previous one with the icon
-        self.assertNotIn(
-            get_html_post_by(user1),
-            self._trim(response.content.decode("utf-8")),
-        )
-        # check that the last post creation writer doesn't contain the icon for user student
-        self.assertNotIn(
-            get_html_post_by(user2),
-            self._trim(response.content.decode("utf-8")),
-        )
+        with self.assertHTML(response, ".topic-last-post") as (html,):
+            extract = parse_html(str(etree.tostring(html)))
+            # control that the instructor's icon is present
+            self.assertFalse(
```

Comment:
when you test that something is NOT in the response, it's better to  test the minimum. If "instructor" in extract is False then it's a better test than testing that this whole sentence is not exactly in the response...

Reply:
that's a good point indeed !